metricbeat/module/kafka/partition/partition.go

```diff
@@ -37,57 +36,60 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch partition stats list from kafka
 func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 
-	if m.client == nil {
+	if m.broker == nil {
```

Comment:
consider allocating/creating m.broker at metricset initialization and  `m.broker.Connected()` to see if broker is connected and reconnect? As `Connected` only checks if a connection object exists, you can try Heartbeat.

Alternatively (to not keep connection open for too long), use `defer m.broker.Close()` and always call `Open`.